cpp/src/arrow/ipc/json-simple.cc

```diff
@@ -644,8 +644,10 @@ class UnionConverter final : public ConcreteConverter<UnionConverter> {
   }
 
   Status AppendNull() override {
-    for (auto& converter : child_converters_) {
-      RETURN_NOT_OK(converter->AppendNull());
+    if (mode_ == UnionMode::SPARSE) {
```

Comment:
Interesting. Is there a way to test for this in `json-simple-test`?

Reply:
This wasn't producing *invalid* dense union arrays (I think) just ones which weren't very dense :) . I could add a test checking that the first `T` in a union from JSON always the first value in the `T` member of the created array

Reply:
I wonder if `ValidateArray` should check that dense arrays really are dense :-) (for example simply by adding lengths of all child arrays). Perhaps open a JIRA if you don't want to do so here.

Reply:
Scratched; would you still like the test? Even if less-than-maximally-dense union arrays have an application, we probably don't want `ArrayFromJSON` to produce any

Reply:
Yes, a basic test would suffice.